src/test/erasure-code/test-erasure-code.sh

```diff
@@ -272,6 +274,140 @@ function TEST_chunk_mapping() {
+    local dir=$1
+
+    # check if osd_pool_erasure_code_subread_all is enabled or not
+    # turn it off if it is enabled
+    #
+    local subread=1
+    CEPH_ARGS='' ./ceph --admin-daemon $dir/ceph-osd.0.asok config get \
+	      "osd_pool_erasure_code_subread_all" | grep "true"
+    if (( $? == 0 )); then
+        subread=0
+	      for id in $(seq 0 10) ; do
+	          kill_osd_daemon $dir $id || return 1
+	          run_osd $dir $id "--osd_pool_erasure_code_subread_all=false" || return 1
+	      done
+    fi
+     
+    local poolname=pool-jerasure
+    local profile=profile-jerasure
+    ./ceph osd erasure-code-profile set $profile \
+        plugin=jerasure \
+        k=4 m=2 \
+        ruleset-failure-domain=osd || return 1
+    ./ceph osd pool create $poolname 12 12 erasure $profile \
+        || return 1
+
+    # inject eio on primary OSD (0), then peer OSD (1)
+    # primary OSD will crash at reading object
+    #
+    for shardid in 0 1; do
+        local objname=obj-eio-$$-$shardid
+    	  local -a initial_osds=($(get_osds $poolname $objname))
+    	  local last=$((${#initial_osds[@]} - 1))
+
+    	  CEPH_ARGS='' ./ceph --admin-daemon $dir/ceph-osd.${initial_osds[$shardid]}.asok config set \
+   	        filestore_debug_inject_read_err true || return 1
+    	  CEPH_ARGS='' ./ceph --admin-daemon $dir/ceph-osd.${initial_osds[$shardid]}.asok injectdataerr \
+	      $poolname $objname $shardid || return 1
+    	  rados_put_get $dir $poolname $objname || return 1
+    	  check_osd_status ${initial_osds[0]} "down" || return 1
+
+    	  # recreate crashed OSD with the same id since I don't know how to restart it :(
+    	  if (( $subread == 0 )); then
+	          if (( $shardid != 0 )); then
+		            run_osd $dir ${initial_osds[0]} "--osd_pool_erasure_code_subread_all=false" || return 1
+	          fi
+    	      run_osd $dir ${initial_osds[$shardid]} "--osd_pool_erasure_code_subread_all=false" || return 1
+    	  else
+	          if (( $shardid != 0 )); then
+		            run_osd $dir ${initial_osds[0]} || return 1
+	          fi
+            run_osd $dir ${initial_osds[$shardid]} || return 1
+    	  fi
+    done
+
+    delete_pool $poolname
+    ./ceph osd erasure-code-profile rm $profile
+}
+
+# this test case is aimed to test the fix of https://github.com/ceph/ceph/pull/2952
+#
+function TEST_rados_get_dataeio_subreadall_jerasure() {
+    local dir=$1
+
+    # check if osd_pool_erasure_code_subread_all is enabled or not
+    # turn it on if it is disabled
+    # skip this case if osd_pool_erasure_code_subread_all is not supported
+    #
+    CEPH_ARGS='' ./ceph --admin-daemon $dir/ceph-osd.0.asok config get \
+	      "osd_pool_erasure_code_subread_all" | grep "error"
+    if (( $? == 0 )); then
+	      echo "Skip this case because osd_pool_erasure_code_subread_all is not supported"
+        return 0
+    fi
+
+    # make sure osd_pool_erasure_code_subread_all is true on every OSD
+    #
+	  for id in $(seq 0 10) ; do
```

Comment:
It is fine in my env. Might be related to different tab setting between mine and yours.

I will try to find a fresh env to rearrange all the scripts.